packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1343,11 +1343,17 @@ class _TabBarState extends State<TabBar> {
       color = Colors.white;
     }
 
-    final bool primaryWithLabelIndicator = widget._isPrimary && indicatorSize == TabBarIndicatorSize.label;
-    final double effectiveIndicatorWeight = theme.useMaterial3 && primaryWithLabelIndicator
-      ? math.max(widget.indicatorWeight, _TabsPrimaryDefaultsM3.indicatorWeight)
+    final double effectiveIndicatorWeight = theme.useMaterial3
+      ? math.max(
+          widget.indicatorWeight,
+          switch (widget._isPrimary) {
+            true  => _TabsPrimaryDefaultsM3.indicatorWeight(indicatorSize),
+            false => _TabsSecondaryDefaultsM3.indicatorWeight,
+          },
+        )
       : widget.indicatorWeight;
     // Only Material 3 primary TabBar with label indicatorSize should be rounded.
+    final bool primaryWithLabelIndicator = widget._isPrimary && indicatorSize == TabBarIndicatorSize.label;
```

Comment:
Nit: Can you convert this to a switch assignment?

Reply:
This looks nice.
```dart
    final bool primaryWithLabelIndicator = switch (indicatorSize) {
      TabBarIndicatorSize.label  => widget._isPrimary,
      TabBarIndicatorSize.tab    => false,
    };
```